packages/dataclass-editor/src/DataClassEditor.tsx

```diff
@@ -10,19 +11,23 @@ import { DataClassMasterToolbar } from './components/dataclass/master/DataClassM
 import { AppProvider } from './context/AppContext';
 import './DataClassEditor.css';
 import { useClient } from './protocol/ClientContextProvider';
-import type { EditorProps } from './protocol/types';
+import type { Data, EditorProps } from './protocol/types';
 import { genQueryKey } from './query/query-client';
+import type { Unary } from './utils/lambda/lambda';
 
 function DataClassEditor(props: EditorProps) {
   const [detail, setDetail] = useState(true);
 
   const [context, setContext] = useState(props.context);
+  const [directSave, setDirectSave] = useState(props.directSave);
   useEffect(() => {
     setContext(props.context);
+    setDirectSave(props.directSave);
```

Comment:
I don't think that you need to have this in this useeffect, as this is only passed as URL parameter right? If the URL changes then also the whole page gets rerendered.

Reply:
I am not sure. I see it done this way in the form editor and config editor as well. Maybe it is needed for the vscode integration or something like that? I can remove it for now and see what happens :shrug:

Reply:
maybe unneeded there as well... not sure
Because this is very an intention setup of this editor behavior and will not change once setup. But let it as it is for now...